src/libraries/Native/Unix/System.Native/pal_io.c

```diff
@@ -1406,10 +1410,28 @@ int64_t SystemNative_GetFileSystemType(intptr_t fd)
     while ((statfsRes = fstatfs(ToFileDescriptor(fd), &statfsArgs)) == -1 && errno == EINTR) ;
     return statfsRes == -1 ? (int64_t)-1 : (int64_t)statfsArgs.f_type;
 #else
-    #error "Platform doesn't support fstatfs"
+    (void)fd; // unused
+    return -1;
```

Comment:
The purpose of the #error here was to know if we were ever building against something that lacked the function we were depending on.  Don't we now silently lose that validation?

Reply:
We surely need that build-time validation. The error is actually moved a few lines below and its condition is changed from `#if !HAVE_STATFS_VFS && !HAVE_STATFS_MOUNT` to have ` && HAVE_NON_LEGACY_STATFS`.

Reply:
That's not the same thing, though.  Let's say we started targeting a new platform that had statvfs but not statfs, yet we were using the code file that invoked SystemNative_GetFileSystemType.  We wouldn't get a compile time error, but we would silently get wrong behavior at run time.

Reply:
Ah, you are right. I was not thinking about association with which partial we will use by default. I have limit the `return -1` case to SunOS platforms.

Reply:
> Would it be better to not add the additional export and instead just do a translation here from file system name to expected ID? Or would that result in duplicating the entire enum we have on the managed side?

I would also prefer to just add a simple mapping to `SystemNative_GetFileSystemType`. We don't need to map all values, we care only about four of them:

https://github.com/dotnet/runtime/blob/28ce2cc5e487f504c8fc91515c41a5fb0e86552c/src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs#L368-L371

Reply:
If we only ever care about four members, should we remove rest from `enum UnixFileSystemTypes`?

Reply:
Ok, I have added the conditions for existing FS types and deleted the additional export. If/when we decide to cleanup `enum UnixFileSystemTypes`, we can cleanup these conditions as well.